### PR TITLE
Clean up player class differences

### PR DESCRIPTION
**Pull request by AJenbo**
So this is my first stab at cleaning up parts of the code. Please tell me if there is stuff that you would like me to do differently or things that I have messed up :P

P.s. this repo dosn't appear to github as a fork of the main repo, this makes it a bit harder to create PR as we need to create a fork of it and not just devilution. Another sugestion would be to have this simply as a branch on the devilution repo.

### CODE REVIEW

**Comment by seritools** [comment]
Sadly github doesn't allow a forked repo in the same account as the original one, so a mirrored repo like this is the next best thing. A branch in the original devilution might be a little bit confusing, with some merge requests being for the master, some for nightly.

File: Source/player.cpp [hunk 1]
```diff
@@ -555,111 +555,107 @@ void __fastcall ClearPlrPVars(int pnum)
...
-	plr[v3]._pNWidth = 96;
-	plr[v3]._pWWidth = 96;
-	plr[v3]._pAWidth = 128;
-	plr[v3]._pHWidth = 96;
-	plr[v3]._pSWidth = 96;
-	plr[v3]._pDWidth = 128;
-	plr[v3]._pBWidth = 96;
-	if ( v6 )
+
+	char pClass = plr[pnum]._pClass;
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/player.cpp]: Thanks, I didn't know where to find these (or how easy it is to use them)

[the remark this answers is not in the document]
Comment (seritools) [reply Source/player.cpp]: Yep, I'm gonna update the contrib info with all the stuff I found over the course of working on devilution :)

**Comment by galaxyhaxz** [comment]
The only reason I try to leave things like `v4->item[0].wCI = *(short *)((char *)&item[0]._iCreateInfo + v9)` in the code for now is that it lets us know there is a signing issue. In this example, item create flags is supposed to be a `WORD` but i had it entered into the decompiler as `signed short`. By seeing this, we know the correct way to fix this is actually by updating the struct file and all signed/unsigned references to `_iCreateInfo`.

PS The reason I had everything listed as `signed` in the decompiler was that it didn't output data correctly if it was signed and in some cases, code.

**Comment by galaxyhaxz** [comment]
@AJenbo the struct file is only 90% correct. It's based entirely off the PSX symbol but many structs were optimized and changed, so some fields are different sizes on the PC release. In many cases `int/BOOL` was converted to `(unsigned) char` on the PSX. So yeah, figuring everything out is hell sometimes :P

**Comment by seritools** [comment]
@AJenbo sure, hit me up on discord :)

For compiling you will need Visual C++ 6 SP3 (or earlier, but definitely not newer - didn't find the exact version yet). That alone is enough to produce functions that are very close to the original, so for cleaning up that's ideal. For producing the final binary you'll also need the linker from VC++ 5 SP3 (Linker version 5.1). I'll update the project/workspace files to VC6 today as well.

@galaxyhaxz In this case, this is just the way the optimized array access works:
`v9 = 368 * v1;`

368 is the struct size, `v1` is the `ii` function parameter, so the line would be translated to `v4->item[0].wCI = item[ii]._iCreateInfo`. Don't think this is because of signed/unsigned in this case, just compiler optimization. ;)

**Comment by galaxyhaxz** [comment]
@seritools In most cases yet. But the `*(short *)` at the beginning is there because it's casting from unsigned to signed, when both fields should be unsigned.

**Comment by seritools** [comment]
@galaxyhaxz Hmm interesting, yeah. Both `wCI` and `_iCreateInfo` are unsigned short, but that cast is `short *`

**Comment by seritools** [comment]
Closing this in favor of smaller PRs as discussed :)

**Pull request closed without merging**